lib/containers/container_images.pm

```diff
@@ -53,10 +54,22 @@ sub build_container_image {
 }
 
 sub test_containered_app {
-    my ($runtime, $container) = @_;
+    my %args       = @_;
+    my $runtime    = $args{runtime};
+    my $dockerfile = $args{dockerfile};
+
+    die "You must define the runtime!"    unless $runtime;
+    die "You must define the Dockerfile!" unless $dockerfile;
+
     my $dir = "/root/containerapp";
-    set_up($dir, "Dockerfile.python");
-    build_img($dir, $runtime);
+
+    # Setup the environment
+    set_up("$dir", $dockerfile);
```

Comment:
One could argue, that `set_up` is a not really descriptive function name and could be renamed to `container_set_up` or something similar.